src/ceph-volume/ceph_volume/api/lvm.py

```diff
@@ -259,7 +259,10 @@ def dmsetup_splitname(dev):
         "--separator=';'", '--nameprefixes', dev
     ]
     out, err, rc = process.call(command)
-    return _splitname_parser(out)
+    if len(out) != 1:
+        logger.error('Unexpected "dmsetup splitname" output for "%s"', dev)
+        return {}
+    return _splitname_parser(out[0])
```

Comment:
this is dangerous because if there is no stdout at all, this would cause an index error. Better to just pass along the full `out` variable

Reply:
if the previous check goes to the parser, then you would need to pass the whole output